kvirt/k3s/__init__.py

```diff
@@ -96,22 +107,38 @@ def create(config, plandir, cluster, overrides):
     if os.path.exists("manifests") and os.path.isdir("manifests"):
         data['files'] = [{"path": "/root/manifests", "currentdir": True, "origin": "manifests"}]
     k = config.k
-    result = config.plan(plan, inputfile='%s/bootstrap.yml' % plandir, overrides=data)
+    bootstrap_overrides = data.copy()
+    bootstrap_install_k3s_args = install_k3s_args.copy()
+    if sdn != 'flannel':
+        bootstrap_install_k3s_args.append("INSTALL_K3S_EXEC='--flannel-backend=none'")
+    bootstrap_install_k3s_args = ' '.join(bootstrap_install_k3s_args)
+    bootstrap_overrides['install_k3s_args'] = bootstrap_install_k3s_args
+    result = config.plan(plan, inputfile='%s/bootstrap.yml' % plandir, overrides=bootstrap_overrides)
     if result['result'] != "success":
         os._exit(1)
+    nodes_overrides = data.copy()
+    nodes_install_k3s_args = install_k3s_args.copy()
+    if sdn != 'flannel':
+        nodes_install_k3s_args.append("INSTALL_K3S_EXEC='--disable-network-policy --no-flannel'")
+    nodes_install_k3s_args = ' '.join(nodes_install_k3s_args)
+    nodes_overrides['install_k3s_args'] = nodes_install_k3s_args
     if masters > 1:
         pprint("Deploying extra masters")
-        if 'name' in data:
-            del data['name']
-        config.plan(plan, inputfile='%s/masters.yml' % plandir, overrides=data)
+        config.plan(plan, inputfile='%s/masters.yml' % plandir, overrides=nodes_overrides)
     firstmasterip, firstmastervmport = _ssh_credentials(k, firstmaster)[1:]
     with open("%s/join.sh" % clusterdir, 'w') as f:
         if api_ip is None:
             api_ip = k.info(firstmaster)['ip']
-        joincmd = "curl -sfL https://get.k3s.io | K3S_URL=https://%s:6443 K3S_TOKEN=%s" % (api_ip, token)
-        if data['sdn'] != "flannel":
-            joincmd += " INSTALL_K3S_EXEC='--no-flannel'"
-        f.write("%s sh -\n" % joincmd)
+        joincmd = "curl -sfL https://get.k3s.io | %s K3S_URL=https://%s:6443 K3S_TOKEN=%s" % (nodes_install_k3s_args,
```

Comment:
the worker installation should also respect the `INSTALL_K3S_CHANNEL` and `INSTALL_K3S_VERSION` vars. To ensure that the version of K3s deployed to masters and workers match at deploy time.

Reply:
the same install_k3s_args are passed to the workers in this call

Reply:
Dang! You have me again - overlooked > https://github.com/karmab/kcli/blob/0ae839b1bd9c96f331049358a7de2b7cdf3e56b8/kvirt/k3s/__init__.py#L120 - that the k3s args are copied.